python/tests/graph/base_test_generators.py

```diff
@@ -0,0 +1,314 @@
+
+TestCase = namedtuple('TestCase', ['bool_result', 'err_msg'])
+TestFn = namedtuple('TestFn', ['test_fn', 'description'])
+_GinInfo = namedtuple('_GinInfo', ['gin', 'description'])
+
+
+class TestGenBase(object):
+    def __init__(self, vec_size=17, test_batch_size=231):
+        # Testing data spec
+        self.vec_size = vec_size
+        self.test_batch_size = test_batch_size
+
+        # TensorFlow graph element names
+        self.input_op_name = 'tnsrOpIn'
+        self.feed_names = []
+        self.output_op_name = 'tnsrOpOut'
+        self.fetch_names = []
+
+        # DataFrame column names
+        self.input_col = 'dfInputCol'
+        self.output_col = 'dfOutputCol'
+
+        # Connecting data from Spark to TensorFlow
+        self.input_mapping = {}
+        self.output_mapping = {}
+
+        # When testing against multiple graph inputs,
+        # derive new names for the DataFrame columns and TensorFlow graph elements.
+        self.reset_iomap(replica=1)
+
+        # The basic stage contains the opaque :py:obj:`TFInputGraph` objects
+        # Any derived that override the :py:obj:`build_input_graphs` method will
+        # populate this field.
+        self.input_graphs = []
+
+        # Construct final test cases, which will be passed to final test cases
+        self.test_cases = []
+
+        # Build a temporary directory, which might or might not be used by the test
+        self._temp_dirs = []
+
+    def make_tempdir(self):
+        """ Create temp directories using this function.
+        At the end of this test object's life cycle, the temporary directories
+        will all be cleaned up.
+        """
+        tmp_dir = tempfile.mkdtemp()
+        self._temp_dirs.append(tmp_dir)
+        return tmp_dir
+
+    def tear_down_env(self):
+        for tmp_dir in self._temp_dirs:
+            shutil.rmtree(tmp_dir, ignore_errors=True)
+
+    def reset_iomap(self, replica=1):
+        self.input_mapping = {}
+        self.feed_names = []
+        self.output_mapping = {}
+        self.fetch_names = []
+
+        if replica > 1:
+            for i in range(replica):
+                colname = '{}_replica{:03d}'.format(self.input_col, i)
+                tnsr_op_name = '{}_replica{:03d}'.format(self.input_op_name, i)
+                self.input_mapping[colname] = tnsr_op_name
+                self.feed_names.append(tnsr_op_name + ':0')
+
+                colname = '{}_replica{:03d}'.format(self.output_col, i)
+                tnsr_op_name = '{}_replica{:03d}'.format(self.output_op_name, i)
+                self.output_mapping[tnsr_op_name] = colname
+                self.fetch_names.append(tnsr_op_name + ':0')
+        else:
+            self.input_mapping = {self.input_col: self.input_op_name}
+            self.feed_names = [self.input_op_name + ':0']
+            self.output_mapping = {self.output_op_name: self.output_col}
+            self.fetch_names = [self.output_op_name + ':0']
+
+    @contextmanager
```

Comment:
this is too complicated and if something goes wrong, you will be hard pressed to understand what is going on.

You should really generate some data, transform this data and compare it against some expected output, and clean up some stuff after that if necessary.